forgejo: 10.0.1 -> 10.0.2

tebriel opened the pull request:
- https://codeberg.org/forgejo/forgejo/src/branch/forgejo/release-notes-published/10.0.2.md
  - Added a test skip for a new test with a network call
- Removed two test skips that are no longer valid (separate commit)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

tebriel committed:
forgejo: 10.0.1 -> 10.0.2

Changelog: https://codeberg.org/forgejo/forgejo/src/branch/forgejo/release-notes-published/10.0.2.md

NyCodeGHG commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 391862 --package forgejo`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>forgejo</li>
    <li>forgejo.data (forgejo.data.data)</li>
  </ul>
</details>

tebriel commented:
```
       > 2025/03/21 13:29:43 ...migrations/common.go:18:WarnAndNotice() [W] PR #3 in <GitlabDownloader http://127.0.0.1:42967 [6590996]/archbuild> has invalid HeadRef: cgtx:review
       > --- FAIL: TestURLAllowedSSH (0.02s)
       >     --- FAIL: TestURLAllowedSSH/Pushmirror_URL (0.00s)
       >         migrate_test.go:128:
       >                Error Trace:    /build/source/services/migrations/migrate_test.go:128
       >                Error:          Received unexpected error:
       >                                migration/cloning from 'git.gay' is not allowed.
       >                Test:           TestURLAllowedSSH/Pushmirror_URL
       > FAIL
       > FAIL   code.gitea.io/gitea/services/migrations 0.758s
       > FAIL
```

looking into why this test fails

pyrox0 commented:
I believe this test should be disabled, because it requires network access and thus would not work inside the NixOS sandbox.

tebriel committed:
forgejo: remove SSH DSA test skips

Tests are no longer valid now that https://codeberg.org/forgejo/forgejo/pulls/4833 is merged into 10.x and backported to 7.x

NyCodeGHG commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 391862 --package forgejo --package forgejo.passthru.tests`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>forgejo.passthru.tests.mysql</li>
    <li>forgejo.passthru.tests.sqlite3</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>forgejo</li>
    <li>forgejo.data (forgejo.data.data)</li>
    <li>forgejo.passthru.tests.postgres</li>
  </ul>
</details>

emilylange commented:
> ssh is segfaulting in forgejo.passthru.tests.mysql ( 😬 )

Lower the build concurrency on your machine.
`nix-build --max-jobs 1` or in case of your `nixpkgs-review` command, use `nixpkgs-review pr 391862 --build-args '--max-jobs 1' --package forgejo --package forgejo.passthru.tests`.

tebriel commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 391862 --package forgejo`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>forgejo</li>
    <li>forgejo.data (forgejo.data.data)</li>
  </ul>
</details>

tebriel commented:
> https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages
> 
> Please add the release notes URL into the commit message.

Done! Thanks I'd missed that in the README.

adamcstephens reviewed (approved):
forgejo.tests on aarch64-linux
/nix/store/a754s3vjb7pg6s0k1c1ii5lspxfx65x7-vm-test-run-forgejo-mysql
/nix/store/xyv6fylqdhahmc95l7v5z5v6i23bbdc0-vm-test-run-forgejo-postgres
/nix/store/j0jk30wf2bn070b7vhlpla33a9wv71vi-vm-test-run-forgejo-sqlite3
